From e83896c8b566a4f1d200b0e9ea8faee2c959b9f4 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Mon, 13 Feb 2023 09:55:48 +0100
Subject: GCC: do not realias RawPtrTraits

Recent change to raw_ptr introducing RawPtrTraits ended up adding
an alias to itself. In GCC that is not allowed as it is redefining
RawPtrTraits.

Bug: 819294
Change-Id: Ia03ff7d44b6887c17ed61da4ff0dd6b8b5d7e22f
---
 base/allocator/partition_allocator/pointers/raw_ptr.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/base/allocator/partition_allocator/pointers/raw_ptr.h b/base/allocator/partition_allocator/pointers/raw_ptr.h
index 86c042caf2556..f587c92e03a5d 100644
--- a/base/allocator/partition_allocator/pointers/raw_ptr.h
+++ b/base/allocator/partition_allocator/pointers/raw_ptr.h
@@ -715,7 +715,6 @@ struct TraitsToImpl {
 // compiler. We have not managed to construct such an example in Chromium yet.
 template <typename T, RawPtrTraits Traits = RawPtrTraits::kEmpty>
 class PA_TRIVIAL_ABI PA_GSL_POINTER raw_ptr {
-  using RawPtrTraits = RawPtrTraits;
   // Type to return from ExtractAsDangling(), which is identical except
   // kMayDangle trait is added (if one isn't there already).
   using DanglingRawPtrType = raw_ptr<T, Traits | RawPtrTraits::kMayDangle>;
-- 
2.37.2

